### PR TITLE
fix Domain sent to apnexus, Index, Criteo, Improve. in AMP

### PR DESCRIPTION
**Pull request by alexmullermrf**

### CODE REVIEW

File: modules/ixBidAdapter.js [hunk 1]
```diff
@@ -475,7 +473,23 @@ export const spec = {
...
+   * @param bidderRequest
+   * @returns {string}
+   */
+  _getPageUrl: function (bidRequest, bidderRequest) {
+    let pageUrl = config.getConfig('pageUrl');
+    if (bidRequest.params.referrer) {
+      pageUrl = bidRequest.params.referrer;
+    } else if (!pageUrl) {
+      pageUrl = bidderRequest.refererInfo.referer;
+    }
```
Comment (oscarmarimrf) [inline modules/ixBidAdapter.js]: Since we want to use always our canonical, I think it's ok to rely on `config.getConfig('pageUrl')`.
```suggestion
```

**Review by oscarmarimrf — CHANGES_REQUESTED** [review]

File: modules/ixBidAdapter.js (hunk 1 above)
Comment (alexmullermrf) [reply modules/ixBidAdapter.js]: Fixed

File: package.json [hunk 2]
```diff
@@ -1,11 +1,12 @@
 {
   "name": "@marfeel/adserver-providers-prebid-library",
-  "version": "3.23.0-pre-marfeel-2",
+  "version": "3.23.1-pre-marfeel-2",
```
Comment (oscarmarimrf) [inline package.json]: We try to keep the original Prebid library version and append a `-marfeel-X` for our changes.

```suggestion
  "version": "3.23.0-pre-marfeel-3",
```

File: src/utils.js [hunk 3]
```diff
@@ -1201,3 +1201,13 @@ export function mergeDeep(target, ...sources) {
 
   return mergeDeep(target, ...sources);
 }
+
+export function _getPageUrl(bidRequest, bidderRequest) {
```
Comment (oscarmarimrf) [inline src/utils.js]: I know we've already updated many adapters but, in order to modify as less library files as possible, I'd move this to [marfeelTools.js](https://github.com/Marfeel/Prebid.js/blob/master/src/marfeelTools.js). What do you think?

File: modules/criteoBidAdapter.js [hunk 4]
```diff
@@ -450,6 +453,10 @@ for (var i = 0; i < 10; ++i) {
 </script>`;
 }
 
+function overwritePublisherURL(data, bidRequests, bidderRequest) {
+  data.publisher.url = utils._getPageUrl(bidRequests, bidderRequest);
```
Comment (oscarmarimrf) [inline modules/criteoBidAdapter.js]:
```suggestion
function overwritePublisherURL(data, bidRequest, bidderRequest) {
  data.publisher.url = utils._getPageUrl(bidRequest, bidderRequest);
```

**Review by oscarmarimrf — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (alexmullermrf) [reply package.json]: Fixed

File: src/utils.js (hunk 3 above)
Comment (alexmullermrf) [reply src/utils.js]: Fixed

File: modules/criteoBidAdapter.js (hunk 4 above)
Comment (alexmullermrf) [reply modules/criteoBidAdapter.js]: Fixed

**Review by oscarmarimrf — APPROVED** [review]

**Review by oscarmarimrf — APPROVED** [review]

**Pull request merged**